### PR TITLE
Test projects automatically inherit app.config

### PR DESCRIPTION
**Pull request by davkean**
Changed it so that all projects inherit a "default"
app.config if one isn't explictly referenced.

### CODE REVIEW

**Comment by davkean** [comment]
Note, I'll follow up deleting app.config's that match the default.

The aim of this PR is to prevent new test projects from running into the EventListener bug - we'll disable appdomain isolation by default.

**Comment by davkean** [comment]
tag @jaredpar @agocke @dotnet/roslyn-infrastructure

File: build/Targets/Roslyn.Toolsets.Xunit.targets [hunk 1]
```diff
@@ -18,4 +18,22 @@
...
 
+  <PropertyGroup>
+    <PrepareForBuildDependsOn>$(PrepareForBuildDependsOn);AddDefaultTestAppConfig</PrepareForBuildDependsOn>
+  </PropertyGroup>
+
+  <!-- Add a default test app.config, if the project doesn't already have one-->
+  <Target Name="AddDefaultTestAppConfig">
+    <PropertyGroup>
+      <_AlreadyHasAppConfig Condition="'%(None.Filename)%(None.Extension)' == 'app.config'">true</_AlreadyHasAppConfig>
+    </PropertyGroup>
```
Comment (davkean) [inline build/Targets/Roslyn.Toolsets.Xunit.targets]: What do you think of this little MSbuild craziness? Nice isn't it. :)

**Comment by davkean** [comment]
@dotnet-bot retest prtest/win/dbg/unit32 please
// Previous failure: http://dotnet-ci.cloudapp.net/job/roslyn_prtest_win_dbg_unit32/3606/
// Retest reason: Network went down during artifacts stage

**Comment by davkean** [comment]
@dotnet-bot retest prtest/win/dbg/unit32 please
// Previous failure: http://dotnet-ci.cloudapp.net/job/roslyn_prtest_win_dbg_unit32/3606/
// Retest reason: Network went down during artifacts stage

**Comment by davkean** [comment]
@dotnet-bot retest prtest/win/dbg/unit64 please
// Previous failure: http://dotnet-ci.cloudapp.net/job/roslyn_prtest_win_dbg_unit32/3606/
// Retest reason: Network went down during artifacts stage

File: build/Targets/Roslyn.Toolsets.Xunit.targets (hunk 1 above)
Comment (jaredpar) [inline build/Targets/Roslyn.Toolsets.Xunit.targets]: What is that?  I dont' understand the none reference.

**Comment by davkean** [comment]
tag @jasonmalinowski

File: build/Targets/Roslyn.Toolsets.Xunit.targets [hunk 2]
```diff
@@ -18,4 +18,22 @@
...
   </PropertyGroup>
 
+  <PropertyGroup>
+    <PrepareForBuildDependsOn>$(PrepareForBuildDependsOn);AddDefaultTestAppConfig</PrepareForBuildDependsOn>
+  </PropertyGroup>
+
+  <!-- Add a default test app.config, if the project doesn't already have one-->
+  <Target Name="AddDefaultTestAppConfig">
+    <PropertyGroup>
+      <_AlreadyHasAppConfig Condition="'%(None.Filename)%(None.Extension)' == 'app.config'">true</_AlreadyHasAppConfig>
```
Comment (jasonmalinowski) [inline build/Targets/Roslyn.Toolsets.Xunit.targets]: The existing MSBuild targets also assume the item type is None?

File: build/Targets/Roslyn.Toolsets.Xunit.targets (hunk 1 above)
Comment (jasonmalinowski) [inline build/Targets/Roslyn.Toolsets.Xunit.targets]: "little MSBuild craziness"? This is 100% normal. :-)

**Comment by jasonmalinowski** [comment]
:+1:

File: build/Targets/Roslyn.Toolsets.Xunit.targets (hunk 2 above)
Comment (davkean) [inline build/Targets/Roslyn.Toolsets.Xunit.targets]: None is what it looks at first, than content. I excluded content at the moment, but I'll add in a followup.

File: build/Targets/Roslyn.Toolsets.Xunit.targets (hunk 1 above)
Comment (davkean) [inline build/Targets/Roslyn.Toolsets.Xunit.targets]: None is the item, ie `<None Include="App.config" />. This is basically a foreach on all of the None items, with an if check to see if its name is app.config.

**Pull request merged**